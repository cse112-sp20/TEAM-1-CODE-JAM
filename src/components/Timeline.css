/* td {
  align-content: center;
  position: relative;
} */

img {
  size: 10px;
  width: 50px;
  height: 50px;
}
#points {
  word-wrap: break-word;
}
table {
  border: none;
}
/* 
.fixed_header {
  width: 400px;
  table-layout: fixed;
  border-collapse: collapse;
}

.fixed_header tbody {
  display: block;
  width: 100%;
  overflow: auto;
  height: 100px;
}

.fixed_header thead tr {
  display: block;
}

.fixed_header thead {
  background: black;
  color: #fff;
}

.fixed_header th,
.fixed_header td {
  padding: 5px;
  text-align: left;
  width: 200px;
} */
/* other */

#animal {
  margin-left: 100px;
}

/* timeline */
#timeline {
  overflow-y: scroll;
  max-height: 544px;
  scrollbar-width: none;
}
#timeline::-webkit-scrollbar {
  display: none;
}
/* Hide scrollbar for IE and Edge */
#timeline {
  -ms-overflow-style: none;
}

.row {
  margin: 0;
}

.timeline {
  position: relative;
}

.timeline .timeline-event {
  position: relative;
  padding-top: 5px;
  padding-bottom: 5px;
}

.timeline .timeline-event .timeline-content {
  position: relative;
  width: calc(50% - 50px);
}

.timeline .timeline-event::before {
  display: block;
  content: "";
  width: 2px;
  height: calc(50% - 30px);
  position: absolute;
  background: #d2d2d2;
  left: calc(50% - 1px);
  top: 0;
}

.timeline .timeline-event::after {
  display: block;
  content: "";
  width: 2px;
  height: calc(50% - 30px);
  position: absolute;
  background: #d2d2d2;
  left: calc(50% - 1px);
  top: calc(50% + 30px);
}

.timeline .timeline-event:first-child::before {
  display: none;
}

.timeline .timeline-event:last-child::after {
  display: none;
}

.timeline .timeline-event:nth-child(even) .timeline-content {
  margin-left: calc(50% + 50px);
}

.timeline .timeline-event:nth-child(odd) .timeline-content {
  margin-left: 0;
}

.timeline .timeline-badge {
  display: block;
  position: absolute;
  width: 40px;
  height: 40px;
  background: #465881;
  top: calc(50% - 20px);
  right: calc(50% - 20px);
  border-radius: 50%;
  text-align: center;
  cursor: default;
}

.timeline .timeline-badge i {
  font-size: 25px;
  line-height: 40px;
}
.card-content {
  font-size: 10px;
}

#myText {
  font-size: 10px;
  word-wrap: break-word;
}
