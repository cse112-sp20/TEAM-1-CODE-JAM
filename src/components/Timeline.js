/*global chrome*/
import React, { Component } from "react";
import "./Timeline.css";

export default class Timeline extends Component {
  constructor(props) {
    super(props);
    this.state = {
      black_listed: [
        "www.youtube.com",
        "www.facebook.com",
        "twitter.com",
        "myspace.com",
      ],
      animals: [
        "alligator",
        "anteater",
        "armadillo",
        "aurochs",
        "axolotl",
        "badger",
        "bat",
        "beaver",
        "buffalo",
        "camel",
        "capybara",
      ],
      leftRightBranch: [],
      urls: [],
    };

    this.createLeftCard = this.createLeftCard.bind(this);
  }
  componentDidMount = () => {
    this.timeline();
    chrome.runtime.onMessage.addListener(this.handleMessage);
    // console.log("dfadfda");
  };

  createRightBranch = (innerHTML, time, update) => {
    let newElement = (
      <div class="timeline-event">
        <div class="card timeline-content">
          <div class="card-image waves-effect waves-block waves-light"></div>
          {/* card */}
          <div class="card-content">
            <span
              id="myText"
              class="card-title activator grey-text text-darken-4"
            >
              {innerHTML}
            </span>
            <p>-20 pts</p>
          </div>
        </div>

        {/* timeline icon */}
        <div class="timeline-badge  red white-text">
          <i class="material-icons">person</i>
        </div>
      </div>
    );
    update
      ? this.state.leftRightBranch.unshift(newElement) // queue
      : this.state.leftRightBranch.push(newElement); // stack
    this.setState({
      leftRightBranch: this.state.leftRightBranch,
    });
  };
  createLeftBranch = (innerHTML, time, update) => {
    {
      /* whole thing event left */
    }
    let newElement = (
      <div class="timeline-event">
        <div class="card timeline-content">
          <div class="card-image waves-effect waves-block waves-light"></div>
          <div class="card-content">
            <span
              id="myText"
              class="card-title activator grey-text text-darken-4"
            >
              {innerHTML}
            </span>
            <p>-24 points</p>
          </div>
        </div>

        <div class="timeline-badge green white-text">
          <i class="material-icons">person</i>
        </div>
      </div>
    );
    update
      ? this.state.leftRightBranch.unshift(newElement) // queue
      : this.state.leftRightBranch.push(newElement); // stack
    this.setState({
      leftRightBranch: this.state.leftRightBranch,
    });
  };

  createLeftCard = (innerHTML, time, update) => {
    let newElement = (
      <div className="row">
        <div className="col s6">
          <div className="card blue-grey lighten-5">
            <div className="card-content white-text">
              <span style={{ color: "black" }} className="card-title">
                {innerHTML}
              </span>
            </div>
          </div>
        </div>
        <div className="col s6">
          <div>
            <h4 style={{ color: "#6886c5" }}>{time}</h4>
          </div>
        </div>
      </div>
    );
    // Differentiate if first time opening timeline
    update
      ? this.state.leftRightBranch.unshift(newElement) // queue
      : this.state.leftRightBranch.push(newElement); // stack
    this.setState({
      leftRightBranch: this.state.leftRightBranch,
    });
  };

  createRightCard = (innerHTML, time, update) => {
    // date.
    let newElement = (
      <div className="row">
        <div className="col s6 ">
          <h4 style={{ color: "#fa9191" }}>{time}</h4>
        </div>
        <div className="col s6">
          <div className="card blue-grey lighten-5">
            <div className="card-content white-text">
              <span style={{ color: "black" }} className="card-title">
                {innerHTML}
              </span>
            </div>
          </div>
        </div>
      </div>
    );
    // Differentiate if first time opening timeline
    update
      ? this.state.leftRightBranch.unshift(newElement) // queue
      : this.state.leftRightBranch.push(newElement); // stack
    this.setState({
      leftRightBranch: this.state.leftRightBranch,
    });
  };

  handleMessage = (msg) => {
    
    // new element for timeline
    if (msg.for === "popup") {
      if (msg.message === "timeline") {
        // let url = msg.url;
        let time = msg.time;
        let url = innerHTML;
        let innerHTML = `Anonymous ${
          this.state.animals[msg.animal % 11]
        }  visited ${msg.url}`;
        msg.flip
          ? this.createLeftBranch(innerHTML, time, 1)
          : this.createRightBranch(innerHTML, time, 1);
        // msg.flip
        //   ? this.createLeftCard(url, time, 1)
        //   : this.createRightCard(url, time, 1);
      }
    }
    console.log(msg);
    return true;
  };

  timeline = async () => {
    //   team name is empty
    let msg = {
      for: "background",
      message: "get timeline",
    };

    let task = new Promise((resolve, reject) => {
      chrome.runtime.sendMessage(msg, function (response) {
        console.log(response);
        resolve(response);
      });
    });
    let data = await task;
    console.log("timeline.js data");
    console.log(data)
    let i = 0;
    data.map((tab) => {
      console.log("timeline.js tab");
      console.log(tab)
      // let url = tab.url;
      let innerHTML = `Anonymous ${this.state.animals[i % 11]}  visited ${
        tab.url
      }`;
      let url = tab.url;
      let time = tab.time;
      let flip = tab.flip;
      flip
        ? this.createLeftBranch(url, time, 0)
        : this.createRightBranch(url, time, 0);
      i++;
      // flip
      //   ? this.createLeftCard(url, time, 0)
      //   : this.createRightCard(url, time, 0);
    });
  };

  render() {
    return (
      <div>
        <div class="container">
          <div id="timeline" class="timeline">
            {this.state.leftRightBranch}
          </div>
        </div>

        {/* <div id="timeline">{this.state.leftRightBranch}</div> */}
      </div>
    );
  }
}
