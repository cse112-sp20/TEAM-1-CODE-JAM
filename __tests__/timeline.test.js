import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";
import Timeline from "../src/components/Timeline";

import chrome from "sinon-chrome";
global.chrome = chrome;
// expect.extend({ toHaveClass });

afterEach(cleanup);

describe("Test if timeline renders without fail", () => {
  test("Render without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Timeline></Timeline>, div);
  });

  test("Testing if table is redered", () => {
    const { getByTestId } = render(<Timeline></Timeline>);
    const timeline = getByTestId("my-timeline");
    const table = getByTestId("my-table");

    expect(timeline).toContainElement(table);
  });
});
