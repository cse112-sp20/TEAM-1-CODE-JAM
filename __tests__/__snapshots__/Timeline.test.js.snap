// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline is set up correctly 1`] = `
<div
  className="row"
  id="myTimeline"
>
  <div
    className="card blue-grey lighten-5"
    id="myCard"
  >
    <div
      className="card-content black-text"
      id="myContent"
    >
      <table
        className="highlight"
        id="myTable"
      >
        <thead>
          <tr>
            <th
              id="myHead"
            >
              Time
            </th>
            <th
              id="myHead"
            >
              Person
            </th>
            <th
              id="myHead"
            >
              Earned
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
